### PR TITLE
`sage.groups.matrix_gps`: Modularization fixes for imports

### PR DESCRIPTION
**Pull request by mkoeppe**
### 📚 Description
We move the `...MatrixGroup_gap` classes to separate modules named `..._gap`.

This is part of:
- #29705 

### 📝 Checklist

- [x] I have made sure that the title is self-explanatory and the description concisely explains the PR.
- [x] I have linked an issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### ⌛ Dependencies

- Depends on #35099 
- Depends on #35119

### CODE REVIEW

**Comment by kwankyu** [comment]
Looks good in the first pass. 

Would you put the boilerplate things like copyright notices in the new files?

**Comment by mkoeppe** [comment]
Done.

File: src/sage/groups/matrix_gps/matrix_group.py [hunk 1]
```diff
@@ -182,10 +184,10 @@ def subgroup(self, generators, check=True):
 
         INPUT:
 
-        - ``generators`` -- a list/tuple/iterable of group elements of self
+        - ``generators`` -- a list/tuple/iterable of group elements of ``self``
         - ``check`` -- boolean (optional, default: ``True``). Whether to check that each matrix is invertible.
 
-        OUTPUT: The subgroup generated by ``generators`` as an instance of FinitelyGeneratedMatrixGroup_gap
+        OUTPUT: The subgroup generated by :meth:`generators` as an instance of :class:`FinitelyGeneratedMatrixGroup_gap`
```
Comment (kwankyu) [inline src/sage/groups/matrix_gps/matrix_group.py:191]: `generators` is an input argument

File: src/sage/groups/matrix_gps/matrix_group_gap.py [hunk 2]
```diff
@@ -231,10 +231,10 @@ def subgroup(self, generators, check=True):
 
         INPUT:
 
-        - ``generators`` -- a list/tuple/iterable of group elements of self
+        - ``generators`` -- a list/tuple/iterable of group elements of ``self``
         - ``check`` -- boolean (optional, default: ``True``). Whether to check that each matrix is invertible.
 
-        OUTPUT: The subgroup generated by ``generators`` as an instance of FinitelyGeneratedMatrixGroup_gap
+        OUTPUT: The subgroup generated by :meth:`generators` as an instance of :class:`FinitelyGeneratedMatrixGroup_gap`
```
Comment (kwankyu) [inline src/sage/groups/matrix_gps/matrix_group_gap.py:238]: `generators` is the input argument

File: src/sage/groups/matrix_gps/matrix_group.py (hunk 1 above)
Comment (mkoeppe) [reply src/sage/groups/matrix_gps/matrix_group.py:191]: Indeed...

File: src/sage/groups/matrix_gps/orthogonal_gap.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
+r"""
+Orthogonal Linear Groups with GAP
+"""
+
```
Comment (kwankyu) [inline src/sage/groups/matrix_gps/orthogonal_gap.py:4]: No boilerplate yet

Comment (mkoeppe) [reply src/sage/groups/matrix_gps/orthogonal_gap.py:4]: Thanks! fixed now

**Review by kwankyu — APPROVED** [review]
Thanks.

**Comment by mkoeppe** [comment]
Thank you!

**Comment by antonio-rojas** [comment]
I'm getting a test failure on Arch with system GAP 4.12.2 after this
```
**********************************************************************
File "/usr/lib/python3.10/site-packages/sage/groups/matrix_gps/finitely_generated_gap.py", line 124, in sage.groups.matrix_gps.finitely_generated_gap.FinitelyGeneratedMatrixGroup_gap.as_permutation_group
Failed example:
    P == Psmaller
Expected:
    False
Got:
    True
**********************************************************************
```
Strangely, the exact same test passed when it was in `finitely_generated.py`

**Comment by kwankyu** [comment]
To examine the failure, I tried to run `tox -e docker-archlinux-latest-standard` but this failed with
```
...
DEPRECATED: The legacy builder is deprecated and will be removed in a future release.
            BuildKit is currently disabled; enable it by removing the DOCKER_BUILDKIT=0
            environment-variable.
...
Step 16/44 : RUN sh -x -c "${BOOTSTRAP}"
 ---> Running in 3cf527175f0d
+ ./bootstrap
bootstrap:39: installing 'm4/sage_spkg_configures.m4'
/sage/build/bin/sage-bootstrap-python: error: none of python python3 python3.12 python3.11 python3.10 python3.9 python3.8 python3.7 python2.7 python3.6 python2 is a suitable Python
...
```
Next I tried to run `ci linux` workflow on my fork of sage. It also failed (actually all flavors of linux failed)

Screen Shot 2023-04-28 at 2 03 40 PM

I have zero experience in testing sage on linux platforms, by the way. What did I wrong?

**Comment by dimpase** [comment]
It's probably a bug in tox file. Anyway, you can just run archlinux in Docker, and build Sage there the usual way.

**Comment by kwankyu** [comment]
Yes. This is a good opportunity for me to try sage on linux.

**Comment by dimpase** [comment]
tagging everything `#random` is a bit lame. One can reset GAP's RNG before a test, like this:
```
gap.Reset(GlobalMersenneTwister,"42")
libgap.Reset(GlobalMersenneTwister,"42")
```

(this is both for pexpect and libgap interfaces)

**Comment by antonio-rojas** [comment]
> tagging everything `#random` is a bit lame. One can reset GAP's RNG before a test, like this:
> 
> ```
> gap.Reset(gap.GlobalMersenneTwister,"42")
> libgap.Reset(libgap.GlobalMersenneTwister,"42")
> ```
> 
> (this is both for pexpect and libgap interfaces - probably just one of these is needed)

That's done in the `Psmaller` constructor already, so the difference must come from somewhere else.

**Comment by antonio-rojas** [comment]
Things get weirder, it depends on the order in which P and Psmaller are defined:
```
sage: imf = libgap.function_factory('ImfMatrixGroup')
sage: GG = imf( 12, 3 )
sage: G = MatrixGroup(GG.GeneratorsOfGroup())
sage: P = G.as_permutation_group()
sage: Psmaller = G.as_permutation_group(algorithm="smaller", seed=6)
sage: P == Psmaller
False
```
vs
```
sage: imf = libgap.function_factory('ImfMatrixGroup')
sage: GG = imf( 12, 3 )
sage: G = MatrixGroup(GG.GeneratorsOfGroup())
sage: Psmaller = G.as_permutation_group(algorithm="smaller", seed=6)
sage: P = G.as_permutation_group()
sage: P == Psmaller
True
```

**Comment by dimpase** [comment]
How about replacing the iffy test with
```
sage: P._libgap_().GroupHomomorphismByImages(Psmaller._libgap_()).Kernel()
Group(())
```

**Comment by antonio-rojas** [comment]
I finally figured out why this happens only when using system gap: the difference is the `radiroot` package, which is not installed by our `gap` system package. Why it has this side effect, I have no idea.

**Pull request merged**